test/CMakeLists.txt

```diff
@@ -1,6 +1,13 @@
 set(BINARY ${PROJECT_BINARY}_test)
 
-file(GLOB_RECURSE TEST_SOURCES LIST_DIRECTORIES false *.h *.cpp)
+file(GLOB_RECURSE TEST_SOURCES LIST_DIRECTORIES false
```

Comment:
Intentional change?

Reply:
yes and a highly necessary one. Before, the files from the actual `src` folder were not included / linked at all. The tests only worked because all of the tested code was just templates or headers which are copied into the cpp files for compilation.